src/system_data.jl

```diff
@@ -208,14 +210,19 @@ function add_time_series!(
     time_series::TimeSeriesData;
     features...,
 )
+    key = nothing
     for component in components
-        add_time_series!(
+        # Component information is not embedded into the key and so it will always be the
+        # same.
+        key = add_time_series!(
```

Comment:
For a bit more robustness we could keep track of the previous key and assert that they're equal — might not be necessary

Reply:
I wanted to do that, but you can't compare immutable structs with Dicts automatically...or at least I didn't find an easy way in a short search. I'm very confident that it is ok.

Reply:
Okay